xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/displayer_page.vm

```diff
@@ -1 +1,23 @@
-#template('displayer_dblist.vm')
+#if ($type == 'edit' || $type == 'search')
+  #if ($field.getProperty('displayType').value == 'input' && $field.getProperty('picker').value == 1)
+    #xpropertySuggestInputDisplayer($field $prefix $name $value)
+  #else
+    $doc.displayEdit($field, $prefix, $object)
+  #end
+#elseif ($type == 'view')
+  #set ($value = $doc.displayView($field, $prefix, $object))
+  #set ($pageDoc = $xwiki.getDocument($value))
+  #if ("$!pageDoc" == "" || $pageDoc.isNew())
+    $value
+  #else
+    <a href="$pageDoc.URL">$pageDoc.getRenderedTitle('plain/1.0')</a>
+  #end
```

Comment:
If you want to cover all cases, it should be:

* no link when the doc is null, which means the current user doesn't have the right to access it
* a link when doc is not null
  * a link to a "missing" doc when the document is new (with proper CSS)
  * a normal link otherwise

Reply:
I also forgot to take care of the case where there are multiple pages.